### ISSUE TITLE
false positive on `NoUnused.Variables` with shadowing

### ISSUE THREAD

**Issue by miyamoen** [jfmengels/elm-review-unused#59]
Hi, I love elm-review. Helpful. Thanks!

**Describe the bug**
NoUnused.Variables reports false positive with shadowing.

**SSCCE (Short Self-Contained Correct Example)**

target code:

```elm
import Html exposing (Html, label, text)

list : List (Html msg)
list =
    [ label [] []
    , Maybe.map
        (\label ->
            text label
        )
        (Just "string")
        |> Maybe.withDefault (text "")
    ]
```

result:

```cli
-- ELM-REVIEW ERROR ------------------- src/UnusedVariableFalsePositive.elm:3:29

(fix) NoUnused.Variables: Imported variable `label` is not used

3| import Html exposing (Html, label, text)
                               ^^^^^

You should either use this value somewhere, or remove it at the location I
pointed at.
```

reports same result with `let in` expression:

```elm
import Html exposing (Html, label, text)

list : List (Html msg)
list =
    [ label [] []
    , let
        label =
            "string"
      in
      text label
    ]
```

Of course, reports with not imported value, too. But `elm make` reports shadowing.

```elm
a : String
a =
    "string"
list : List String
list =
    [ a
    , Maybe.map (\a -> a)
        (Just "shadowing")
        |> Maybe.withDefault ""
    ]
```

reports:

```cli
(fix) NoUnused.Variables: Top-level variable `a` is not used

29| a : String
30| a =
    ^
31|     "string"

You should either use this value somewhere, or remove it at the location I
pointed at.
```

`elm make` reports:

```cli
-- SHADOWING ------------------------------- src/UnusedVariableFalsePositive.elm

The name `a` is first defined here:

30| a =
    ^
But then it is defined AGAIN over here:

37|     , Maybe.map (\a -> a)
                      ^
Think of a more helpful name for one of them and you should be all set!

Note: Linters advise against shadowing, so Elm makes “best practices” the
default. Read <https://elm-lang.org/0.19.1/shadowing> for more details on this
choice.
```

**Expected behavior**
no report.
But I avoid shadowing, change variable names. Improved code.

**Additional context**
`elm-review` CLI version: 2.6.1
`"jfmengels/elm-review": "2.6.1"`
`"jfmengels/elm-review-unused": "1.1.19"`

**Comment by jfmengels** [comment]
Hi @miyamoen

Thank you for the helpful issue! I think I have fixed the issue on `master`. Could you try out the following command to see if my fix solved yur issue? If so, I will release the fix ASAP :) 

```bash
elm-review --template jfmengels/elm-review-unused/preview --rules NoUnused.Variables
```

(You may need to use `--ignore-dirs` / `--ignore-files` if you ignored parts of your project in your review configuration)

**Comment by miyamoen** [comment]
My code passed new NoUnused.Variables rule. Moreover it found some other unused imported functions.
Thank you for your works!

**Comment by jfmengels** [comment]
Perfect, thank you @miyamoen!
I have released the fix in v`1.1.20`.
